content/blog/tyranny-of-nits/index.md

```diff
@@ -0,0 +1,208 @@
+
+So far, none of this is very controversial.
+If you've been a software developer for long at all, you've been on the receiving end of nits and scope creep,
+or watched it block a feature that you've desperately awaited as a functioning PR just quietly dies.
+It's really tempting, and really easy, to just blame the reviewer here.
+If they just chilled, just let the nits slide, just fixed them themselves, didn't scope creep: then everything would be okay!
+
+But that's not a real solution for anyone other than "the nitpicking reviewer".
+Around here, **we believe in systematic solutions to systematic problems.**
+And if reading this post has left you with as many bitter memories of [great ideas that may never be](https://soasis.org/posts/a-mirror-for-rust-a-plan-for-generic-compile-time-introspection-in-rust/) as writing it left in me,
+you can be assured this is a systematic problem.
+
+## Automation for humans
+
+Axiomatically, there are two ways to solve nitpicking in code review.
+First, reduce the amount of nits that need picking.
+Ensure that code that comes in is free of obvious trivial mistakes.
+Linters and formatters are great for that!
+
+But just as importantly: invest in paying off your tech debt and clarifying your architecture.
+If it's *obvious* how to do something, you won't have to waste everyone's time explaining that someone did something the wrong way.
+Get those eager new contributors to help, seriously.
+
+Secondly, you can reduce the maount of nits that are picked.
+I don't actually think that telling reviewers "don't worry about this stuff" is generally a useful strategy.
+Instead, get them to focus on splitting apart their "would be nice" feedback from "this must be fixed" feedback.
+
+If you want to implement *any* of these changes effectively you need automation: whether that's CI for the machines or written guidelines for humans.
+[I gave a whole talk about this](https://www.youtube.com/watch?v=u3PJaiSpbmc): go watch it!
+
+## A taxonomy of code concerns
+
+Nits are in the eye of the beholder (we would make a joke here, but then [WotC](https://en.wikipedia.org/wiki/Wizards_of_the_Coast) would sue us for [trademark infringement](https://en.wikipedia.org/wiki/Beholder_(Dungeons_%26_Dragons))).
+Ultimately it comes down to the question of "what should be blocking".
+
+Let's make the **Leafwing Taxonomy of Code Concerns** (your PMs are more likely to use it if it has a fancy title).
+Years of painstaking fieldwork and sample collection have led me to the identification of four categories.
+
+1. **Clear mistakes:** the code doesn't work as intended.
+2. **Future work:** there's something cool that's related that we could or should do!
+3. **Polish problems:** documentation, formatting, micro-scale code quality problems.
+4. **Compromise conflict:** the change sacrifices some desirable property in favor of another one, but the reviewer doesn't agree that it's the right choice.
```

Comment:
I'd actually include a 5th: architecture and tech debt issues. This is stuff like "this new thing deviates from the standard pattern, please make it more consistent" and "this architecture will be very hard to refactor" and other such things. They're like, higher level code quality concerns, usually brought up by more senior folks, and folks who have a lot of familiarity with the project.